File: Week1/problem1.cs
/*
Language: c#
ID: 60572
QLink: https://quera.org/course/assignments/60572/problems
Author: @Amirkhaskar
*/

using System;
using System.Collections.Generic;

public class Program {
    public static int RomanToInt(string s) {
        Dictionary<char, int> romanDict = new Dictionary<char, int>() {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };
        
        int result = 0;
        int prevValue = 0;
        
        foreach (char c in s) {
            int currValue = romanDict[c];
            result += currValue;
            
            if (prevValue < currValue) {
                result -= 2 * prevValue;
            }
            
            prevValue = currValue;
        }
        
        return result;
    }
    
    public static void Main(string[] args) {
        string n = Console.ReadLine();
        Console.WriteLine(RomanToInt(n));
    }
}

File: Week1/problem1.py
# Language: python
# ID: 60572
# QLink: https://quera.org/course/assignments/60572/problems
# Author: @Amirkhaskar

def romanToInt(s):
    roman_dict = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    }
    
    result = 0
    prev_value = 0
    
    for char in s:
        curr_value = roman_dict[char]
        result += curr_value
        
        if prev_value < curr_value:
            result -= 2 * prev_value
        
        prev_value = curr_value
    
    return result

# Reading input
n = input().strip()

# Output
print(romanToInt(n))
